pkg/scheduler/framework/register.go

```diff
@@ -0,0 +1,37 @@
+/*
```

Comment:
I think we should move `registry.go` to `framework/` and only move the definition of `type PluginFactory = ...` to `framwork/v1alpha1/interface.go`.

Reply:
I tried that, there is also an import cycle:
* `framwork/registry.go` uses `v1alpha1.PluginFactory`, then `framework` will import `framwork/v1alpha1`.
* `framwork/v1alpha1/framework.go` uses `framwork.Registry`, then `framwork/v1alpha1` will import `framework`

There is a lot of relationships between these types and interfaces: `Registry` -> `PluginFactory` -> `FrameworkHandle` -> `WaitingPod`, so we need to move `PluginFactory`, `FrameworkHandle`, `WaitingPod` and other related types/interfaces to a separate package(e.g. `framework/v1alpha1/types`). How about this? It might be better to keep it as the the PR is now.

Reply:
@bsalamat Any suggestion for it?

Reply:
> When we created the configuration for plugins, we discussed the order of their execution. We tried to capture the final decision in [this comment](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/types.go#L143). Here is a snippet:
> 
> ```
> Enabled plugins are called in the order specified here, after default plugins. If they need to
> be invoked before default plugins, default plugins must be disabled and re-enabled here in desired order.
> ```

I met the same problem when converting the binding function to a plugin, I think we could add another field `Priority` in the `Plugin` struct to solve this problem, and I created an issue https://github.com/kubernetes/kubernetes/issues/82050 for this proposal, 

```go
type Plugin struct {
	// Name defines the name of plugin
	Name string
	// Weight defines the weight of plugin, only used for Score plugins.
	Weight int32
	Priority int32
}
```

Reply:
@bsalamat @ahg-g Yes, the function is empty now, it aims to solve import cycle for PR of https://github.com/kubernetes/kubernetes/issues/80304(converting default sort plugin). I split it to two PRs because others might also need this PR to convert other default plugins. I'll send the PR for #80304, then we could figure out the problem and solution.

Reply:
Just sent a PR https://github.com/kubernetes/kubernetes/pull/82204 for #80304 that depends on this PR to explain its purpose.

Reply:
A good solution to move forward with for now is to create a ```default_registry.go``` under framework/plugins/, the file would include a function named NewDefaultRegistry similar to the one defined here that returns a registry instance with plugins created under framework/plugins.

Reply:
Sorry for misunderstanding, it is not introduced by your approach. 

When I implemented #82204, I found another import cycle, I just added a note in that PR. I have resolved in another way, although it smells a little bad.

Reply:
> Sorry for misunderstanding, it is not introduced by your approach.
> 
> When I implemented #82204, I found another import cycle, I just added a note in that PR. I have resolved in another way, although it smells a little bad.

That import cycle is introduced by that issue is related to Plugin configuration, which is a different story. Plugin configuration shouldn't be done in framework.go.